powershell.html.markdown

```diff
@@ -13,116 +14,348 @@ Nearly all examples below can be a part of a shell script or executed directly
 in the shell.
 
 A key difference with Bash is that it is mostly objects that you manipulate
-rather than plain text.
+rather than plain text. After years of evolving, it resembles Python a bit.
 
 [Read more here.](https://docs.microsoft.com/powershell/scripting/overview)
 
-If you are uncertain about your environment:
+Powershell as a Language:
 
 ```powershell
-Get-ExecutionPolicy -List
-Set-ExecutionPolicy AllSigned
-# Execution policies include:
-# - Restricted: Scripts won't run.
-# - RemoteSigned: Downloaded scripts run only if signed by a trusted publisher. 
-# - AllSigned: Scripts need to be signed by a trusted publisher.
-# - Unrestricted: Run all scripts.
-help about_Execution_Policies # for more info
 
-# Current PowerShell version:
-$PSVersionTable
-```
+# Single line comments start with a number symbol.
 
-Getting help:
+<#
+  Multi-line comments
+  like so
+#>
 
-```powershell
-# Find commands
-Get-Command about_* # alias: gcm
-Get-Command -Verb Add
-Get-Alias ps
-Get-Alias -Definition Get-Process
+####################################################
+## 1. Primitive Datatypes and Operators
+####################################################
+
+# Numbers
+3 # => 3
+
+# Math
+1 + 1   # => 2
+8 - 1   # => 7
+10 * 2  # => 20
+35 / 5  # => 7.0
+
+# Powershell uses banker's rounding
+# Meaning [int]1.5 would round to 2 but so would [int]2.5
+# division always returns a float. You must cast result to [int] to round
+[int]5 / [int]3       # => 1.66666666666667
+[int]-5 / [int]3      # => -1.66666666666667
+5.0 / 3.0   # => 1.66666666666667
+-5.0 / 3.0  # => -1.66666666666667
+[int]$result = 5 / 3 # => 2
+
+# Modulo operation
+7 % 3  # => 1
+
+# Exponentiation requires longform or the built-in [Math] class
+[Math]::Pow(2,3)  # => 8
+
+# Enforce order of operations with parentheses
+1 + 3 * 2  # => 7
+(1 + 3) * 2  # => 8
+
+# Boolean values are primitives (Note: the $)
+$True  # => True
+$False  # => False
+
+# negate with !
+!$True   # => False
+!$False  # => True
+
+# Boolean Operators
+# Note "-and" and "-or" usage
+$True -and $False  # => False
+$False -or $True   # => True
+
+# True and False are actually 1 and 0 but only support limited arithmetic
+# However, casting the bool to int resolves this
+$True + $True # => 2
+$True * 8    # => '[System.Boolean] * [System.Int32]' is undefined
+[int]$True * 8 # => 8
+$False - 5   # => -5
+
+# Comparison operators look at the numerical value of True and False
+0 -eq $False  # => True
+1 -eq $True   # => True
+2 -eq $True   # => False
+-5 -ne $False # => True
+
+# Using boolean logical operators on ints casts them to booleans for evaluation
+# but their non-cast value is returned
+# Don't mix up with bool(ints) and bitwise and/or (&,|)
+[bool](0)     # => False
+[bool](4)     # => True
+[bool](-6)    # => True
+0 -and 2     # => 0
+-5 -or 0     # => -5
```

Comment:
My output is different

Reply:
Needed to change to -band/-bor for posh. Changed in 998249d